src/components/structures/GroupView.js

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2017 Vector Creations Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import MatrixClientPeg from '../../MatrixClientPeg';
+import sdk from '../../index';
+import { sanitizedHtmlNode } from '../../HtmlUtils';
+import { _t } from '../../languageHandler';
+
+
+module.exports = React.createClass({
+    displayName: 'GroupView',
+
+    propTypes: {
+        groupId: React.PropTypes.string.isRequired,
+    },
+
+    getInitialState: function() {
+        return {
+            summary: null,
+            error: null,
+        };
+    },
+
+    componentWillMount: function() {
+        this._loadGroupFromServer(this.props.groupId);
+    },
+
+    componentWillReceiveProps: function(new_props) {
+        if (this.props.groupId != new_props.groupId) {
+            this.setState({
+                summary: null,
+                error: null,
+            })
+            this._loadGroupFromServer(new_props.groupId);
+        }
+    },
+
+    _loadGroupFromServer: function(groupId) {
+        MatrixClientPeg.get().getGroupSummary(groupId).done((res) => {
+            this.setState({
+                summary: res,
+                error: null,
+            });
+        }, (err) => {
+            this.setState({
+                summary: null,
+                error: err,
+            });
+        });
+    },
+
+    render: function() {
+        const BaseAvatar = sdk.getComponent("avatars.BaseAvatar");
+        const Loader = sdk.getComponent("elements.Spinner");
+
+        if (this.state.summary === null && this.state.error === null) {
+            return <Loader />;
+        } else if (this.state.summary) {
+            const summary = this.state.summary;
+            let avatarUrl = null;
+            if (summary.profile.avatar_url) {
+                avatarUrl = MatrixClientPeg.get().mxcUrlToHttp(summary.profile.avatar_url);
```

Comment:
What exactly is it you're worries about here? Getting the matrix client at render time because sometimes there might not be one?